File: SFA-Client/src/app/_models/order.ts
import { OrderItemBatch } from './orderItemBatch';

export class Order {
  id: number;
  shopId: number;
  shopName: string;
  userId: number;
  salesRepName: string;
  orderCode: string;
  totalAmount: number;
  isComplete: boolean;
  completedDate: Date;
  orderedDate: Date;
  isEdit: boolean;
  editedDate: Date;
  editedUserId: number;
  isCancel: boolean;
  canceledDate: Date;
  canceledUserId: number;
  canceledReason: string;
  isSync: boolean;
  syncedDate: Date;
  loginEmail: string;
  orderItemBatches: OrderItemBatch[];

  constructor(order) {
    this.id = order.id;
    this.shopId = order.shopId;
    this.shopName = order.shopName;
    this.userId = order.userId;
    this.salesRepName = order.salesRepName;
    this.orderCode = order.orderCode;
    this.totalAmount = order.totalAmount;
    this.isComplete = order.isComplete;
    this.completedDate = order.completedDate;
    this.orderedDate = order.orderedDate;
    this.isEdit = order.isEdit;
    this.editedDate = order.editedDate;
    this.editedUserId = order.editedUserId;
    this.isCancel = order.isCancel;
    this.canceledDate = order.canceledDate;
    this.canceledUserId = order.canceledUserId;
    this.canceledReason = order.canceledReason;
    this.isSync = order.isSync;
    this.syncedDate = order.syncedDate;
    this.orderItemBatches = order.orderItemBatches;
  }
}

File: SFA-API/Domain/Order.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Order
    {
        public int Id { get; set; }
        public int ShopId { get; set; }
        public int UserId { get; set; }
        public string OrderCode { get; set; }
        public decimal TotalAmount { get; set; }
        public bool IsComplete { get; set; }
        public DateTime? CompletedDate { get; set; }
        public DateTime OrderedDate { get; set; }
        public bool IsEdit { get; set; }
        public DateTime? EditedDate { get; set; }
        public int EditedUserId { get; set; }
        public bool IsCancel { get; set; }
        public DateTime? CanceledDate { get; set; }
        public int CanceledUserId { get; set; }
        public string CanceledReason { get; set; }
        public bool IsSync { get; set; }
        public DateTime? SyncedDate { get; set; }
        public virtual ICollection<OrderItemBatch> OrderItemBatches { get; set; }
        public virtual Shop Shop { get; set; }
        public virtual UserDetail UserDetail { get; set; }
    }
}